### PR TITLE
Fix panic from syslog trying to log a nil error with Error

### PR DESCRIPTION
**Pull request by mmlb**
## Description

The syslog package was logging severities 0-3 (emerg, alert, crit, err) using `log.Error`. This ends up passing a nil error to `Error()` which `pkg/log`'s rollbar tries to call `.Error` on. And boom panic.

This is PR handles this by not logging as `Error` in the syslog package, as its not really a boots error anyway, and by updating to version of `packethost/pkg/log` that returns from `Error` if error is nil.

## How Has This Been Tested?

Ran locally in dev testing. Will be tried in lab/staging.

## How are existing users impacted? What migration steps/scripts do we need?

No panics due to syslog messages.

### CODE REVIEW

**Review by markyjackson-taulia — APPROVED** [review]

File: syslog/receiver.go [hunk 1]
```diff
@@ -11,9 +11,17 @@ import (
 )
 
 var (
-	syslogAddr = conf.SyslogBind
-
-	loggerFuncs map[byte]func(...interface{})
+	syslogAddr  = conf.SyslogBind
+	loggerFuncs = map[byte]func(...interface{}){
```
Comment (jacobweinstock) [inline syslog/receiver.go:15]: this appears to only be used at line 122. maybe don't need a package variable https://github.com/tinkerbell/boots/blob/e58c69257db494a216b41813b27f0d230dd195c6/syslog/receiver.go#L122

File: Makefile [hunk 2]
```diff
@@ -12,31 +12,32 @@ SHELL := bash
 .SHELLFLAGS := -o pipefail -euc
 
 binary := boots
```
Comment (jacobweinstock) [inline Makefile:14]: looks like this isn't used anymore, might consider removing it

File: Makefile [hunk 3]
```diff
@@ -66,36 +67,24 @@ ipxe/ipxe/build/bin-arm64-efi/snp.efi ipxe/ipxe/build/bin-x86_64-efi/ipxe.efi ip
...
+golangci-lint: ## Run golangci-lint 
 	@echo be sure golangci-lint is installed: https://golangci-lint.run/usage/install/
 	golangci-lint run
 
-.PHONY: validate-local
-validate-local: vet go-test goimports golangci-lint # validate-local runs all the same validations and tests that CI run
-	
+validate-local: vet coverage goimports golangci-lint ## Runs all the same validations and tests that run in CI
 	
+help: ## Print this help
```
Comment (jacobweinstock) [inline Makefile:89]: yeah!!

**Review by jacobweinstock — DISMISSED** [review]
couple of comments but nothing to block on

File: Makefile (hunk 2 above)
Comment (mmlb) [reply Makefile:14]: yep thanks find/replace doesn't make unnecessary things go away :D

File: Makefile (hunk 3 above)
Comment (mmlb) [reply Makefile:89]: :highfive:

File: syslog/receiver.go (hunk 1 above)
Comment (mmlb) [reply syslog/receiver.go:15]: very true, will drop

**Comment by mmlb** [comment]
@jacobweinstock good call on the `loggerFuncs`, I uh couldn't help myself and needed to make the other maps go away too, PTAL.

**Review by jacobweinstock — APPROVED** [review]

**Review by jacobweinstock — APPROVED** [review]
do we need a `go generate` call in the `Makefile` somewhere now?

**Review by markyjackson-taulia — APPROVED** [review]

**Comment by markyjackson-taulia** [comment]
Merging in known borked state and PR will following to add `go generate` to makefile

**Pull request merged**